servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpRequest.java

```diff
@@ -245,7 +245,7 @@ public <T> HttpRequest payloadBody(final T pojo, final HttpSerializer2<T> serial
     public HttpHeaders trailers() {
         if (trailers == null) {
             trailers = original.payloadHolder().headersFactory().newTrailers();
-            original.transform(this);
+            original.transform(this);   // Invoke "transform" to set PayloadInfo.mayHaveTrailers() flag
```

Comment:
this seemed odd to me and was one thing I had trouble grokking when I was trying to track this down.

Reply:
I agree 😞 
Unfortunately, we currently don't have a nice way to update `PayloadInfo` flags without invoking transformation that will do it for us. Maybe we refactor in the future.